app/javascript/utils/EagerPagination.js

```diff
@@ -3,6 +3,8 @@ import { slice } from 'lodash'
 // Note: This needs to match the page size defined on the server in soql_query_builder.rb.
 const SERVER_PAGE_SIZE = 100
 
+const MAX_SERVER_LIMIT = 2000
```

Comment:
It's technically 2100 (max offset is 2000, but we can get the last page so we can show 2100 results)